clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -46,4 +57,53 @@ public abstract class AzureBaseClient {
       throw new RuntimeException("Failed to create Azure Resource Management Service", e)
     }
   }
+
+  public String getAzureRESTJson(AzureCredentials creds, String baseUrl, String targetUrl, List<String> queryParameters){
+    String result = null
+    String url = AzureUtilities.getAzureRESTUrl(subscriptionId, baseUrl, targetUrl, queryParameters)
+
+    try {
+      // create HTTP Client
+      def config = buildConfiguration(creds)
+      HttpClientBuilder httpClientBuilder = config.create(HttpClientBuilder.class)
```

Comment:
Also, reason you're using httpclient vs. retrofit or OkHttp? (later being more standard within Spinnaker)